libkbfs/folder_block_ops.go

```diff
@@ -1546,8 +1546,18 @@ func (fbo *folderBlockOps) Truncate(
 func (fbo *folderBlockOps) IsDirty(lState *lockState, file path) bool {
 	fbo.blockLock.RLock(lState)
 	defer fbo.blockLock.RUnlock(lState)
-	return fbo.config.DirtyBlockCache().IsDirty(
-		fbo.id(), file.tailPointer(), file.Branch)
+	// Definitely dirty if a block is dirty.
+	if fbo.config.DirtyBlockCache().IsDirty(
+		fbo.id(), file.tailPointer(), file.Branch) {
+		return true
+	}
+
+	// The deCache entry could still be dirty, if a file had an
+	// attribute set (like mtime or exec) after the file was removed.
+	// Still count the file as dirty in that case; most likely, the
+	// next call will just clear out the cache info.
```

Comment:
next call to what? perhaps `next call to syncLocked will just clear out the cache info (see comments there)`